torchx/mix.exs

```diff
@@ -12,8 +12,9 @@ defmodule Torchx.MixProject do
       elixir: "~> 1.12-dev",
       deps: deps(),
       docs: docs(),
-      compilers: [:elixir_make] ++ Mix.compilers(),
-      elixirc_paths: elixirc_paths(Mix.env())
+      compilers: [:torchx, :elixir_make] ++ Mix.compilers(),
```

Comment:
Here is what I have been thinking:

```suggestion
      compilers: compilers() ++ Mix.compilers(),
```

where:

```elixir
  defp compilers do
    if System.get_env("LIBTORCH_DIR"), do: [:elixir_make], else: [:torchx, :elixir_make]
  end

  defp aliases do
    [
      "compile.torchx": &compile/1
    ]
  end

  defp compile(args) do
    cache_dir = Path.join(__DIR__, "cache")
    # TODO: This directory name must also include the libtorch version.
    # Otherwise if someone depends on torchx as a git dependency, they
    # can upgrade the version and we will think it is cached but it is actually old.
    libtorch_dir = Path.join(cache_dir, "libtorch")

    if "--force" in args do
      File.rm_rf!(cache_dir)
    end

    unless File.dir?(libtorch_dir) do
      install_libtorch(cache_dir, libtorch_dir)
    end
  end
```

If people setup a bad LIBTORCH_DIR, then it is on them. :)

Reply:
@josevalim I've commited this + a file which indicates the default libtorch version, to be used for the default zip and dir by both the Makefile and mix.exs. Let me know if I should revert this last addition :)